### PR TITLE
Development

### PR DESCRIPTION
**Pull request by suretrust**
Kindly review.

### CODE REVIEW

File: src/apiData.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        .then(res => {
+            if (res.message) {
+                error.style.display = "block";
+                error.innerText = `${capitalizeFirstLetter(
+                    res.message
+                )}, please try again.`;
+                weatherDetails.innerHTML = ``;
+            } else {
+                error.style.display = "none";
+                weatherDetails.innerHTML = `<div class="animated rotateIn card  bg-secondary text-white my-3 mx-md-5 shadow-lg">
```
Comment (BalintMendli) [inline src/apiData.js]: This module is doing too much. DOM related logic should be in a separate module. Please move it to a separate module file.

File: src/apiData.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+                                                        </div>
+                                                        <div class="my-auto">
+                                                            ${moment.unix(res.dt).utcOffset(res.timezone / 60).format('LT')}
+                                                        </div>
+                                                    </div>
+                                                </div>
+                                            </div>`;
+            }
+        })
+        .catch(error => error);
```
Comment (BalintMendli) [inline src/apiData.js]: Error handling should happen here in the `catch` block. There could be errors when you don't get a response back and you don't handle those now.

File: src/conversion.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+    const celsiusToFahrenheit = (temp) => {
+        return ((temp * (9 / 5)) + 32).toFixed(1);
+    }
+
+    return {
+        celsiusToFahrenheit,
+        fahrenheitToCelsius
+    }
+})();
```
Comment (BalintMendli) [inline src/conversion.js]: Why do we need an IIFE here? It is already a module. We could have just this: 
```
const fahrenheitToCelsius = (temp) => {
  return ((temp - 32) * (5 / 9)).toFixed(1);
}

const celsiusToFahrenheit = (temp) => {
  return ((temp * (9 / 5)) + 32).toFixed(1);
}

export { celsiusToFahrenheit, fahrenheitToCelsius }
```

File: src/clickListeners.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+            const weatherDetails = document.getElementById("weather-details");
+            error.style.display = "none";
+
+            weatherDetails.innerHTML = `<div class="d-flex justify-content-center mt-5">
+                                            <div class="spinner-border" role="status">
+                                                <span class="sr-only">Loading...</span>
+                                            </div>
+                                        </div>`
+
+            apiData(`http://api.openweathermap.org/data/2.5/weather?q=${getInputCity()}&units=imperial&APPID=e43078cb1a48713be6d68ea7035e03f0`);
```
Comment (BalintMendli) [inline src/clickListeners.js]: Use the secure `https` link here otherwise it won't work on the deployed version.

**Review by BalintMendli — CHANGES_REQUESTED** [review]
Hi @suretrust ,

Wonderful work on the project. 👏 Nice UI, fetching and displaying data works as it should. Unit conversion works as well.

We just need to make some minor tweaks to make this perfect.

Please see my code comments below and make the requested changes. When it's done submit another Code Review Request.

If you happen to have any questions or concerns do not hesitate to contact me on Slack or here in the comments.

**Comment by suretrust** [comment]
All changes made as suggested. Kindly review.

File: src/apiData.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+        .then(response => {
+            hide('error');
+            showWeatherBox(
+                response.name,
+                response.sys,
+                response.dt,
+                response.timezone,
+                response.weather,
+                response.wind,
+                response.main
```
Comment (BalintMendli) [inline src/apiData.js]: Consider using destructuring here for the response object.

**Review by BalintMendli — APPROVED** [review]
Hi @suretrust ,

All good now, this is a

## 👍 


You are a Promise wizard now 😉 

You can go ahead and merge the project.

_P.S. Please use [Git flow](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59882084/)  for your next projects._

**Pull request merged**